### PR TITLE
feat: bonjour options

### PR DESCRIPTION
**Pull request by anshumanv**
- [ ] This is a **bugfix**
- [x] This is a **feature**
- [ ] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?
wip

### Motivation / Use-Case
Allows options from bonjour for more configuration - https://github.com/watson/bonjour#initializing

### Breaking Changes
no

### CODE REVIEW

**Comment by anshumanv** [comment]
@alexander-akait tests for object bonjour config and use `https` when passed in `devServer` options, anything else?

File: lib/options.json [hunk 1]
```diff
@@ -140,7 +140,14 @@
...
   "properties": {
     "bonjour": {
-      "type": "boolean",
+      "anyOf": [
+        {
+          "type": "boolean"
+        },
+        {
+          "type": "object"
+        }
```
Comment (snitin315) [inline lib/options.json:149]: Need to add `"properties"` as well. https://www.npmjs.com/package/bonjour#var-service--bonjourpublishoptions

Comment (anshumanv) [reply lib/options.json:149]: Could be problematic if they change in the package, we'll have to update manually and do release everytime

Comment (snitin315) [reply lib/options.json:149]: I would prefer to keep it as `boolean` only and respect `options.https`

Comment (snitin315) [reply lib/options.json:149]: It also seems this package doesn't go under changes anymore. Last publish 5 years ago.

Comment (anshumanv) [reply lib/options.json:149]: we need to support all options

Comment (anshumanv) [reply lib/options.json:149]: hm then we can have properties, lets do it

File: lib/options.json [hunk 2]
```diff
@@ -145,7 +145,34 @@
...
+              "type": "array",
+              "items": {
+                "type": "string"
+              }
+            },
+            "txt": {
+              "type": "object"
+            }
+          },
+          "additionalProperties": false
```
Comment (alexander-akait) [inline lib/options.json:175]: I still think we don't need this, new options can be added and we don't need do extra work here it is 3rd party code

Comment (anshumanv) [reply lib/options.json:175]: https://github.com/webpack/webpack-dev-server/pull/3202#discussion_r616536873

Comment (anshumanv) [reply lib/options.json:175]: hasn't done any changes since 5y

Comment (alexander-akait) [reply lib/options.json:175]: Tomorrow all can be changed

Comment (anshumanv) [reply lib/options.json:175]: lets avoid it then 😮

Comment (anshumanv) [reply lib/options.json]: done

Comment (snitin315) [reply lib/options.json]: yes, lets keep only `boolean` and set `https` when `options.https` else `http`

Comment (anshumanv) [reply lib/options.json]: https://github.com/webpack/webpack-dev-server/pull/3202#discussion_r616536946 we need support all options

Comment (alexander-akait) [reply lib/options.json]: `protocol (string, optional) - udp or tcp (default)` and `txt (object, optional) - a key/value object to broadcast as the TXT record` can be useful too in some cases

File: lib/utils/runBonjour.js [hunk 3]
```diff
@@ -1,14 +1,15 @@
...
   const os = require('os');
-
+  const { https, port } = options;
   bonjour.publish({
     name: `Webpack Dev Server ${os.hostname()}:${port}`,
     port,
-    type: 'http',
+    type: https ? 'https' : 'http',
     subtypes: ['webpack'],
+    ...(typeof options.bonjour === 'object' && options.bonjour),
```
Comment (alexander-akait) [inline lib/utils/runBonjour.js:12]: No need extra check (`...undefined` works fine), just use `...options.bonjour`

Comment (anshumanv) [reply lib/utils/runBonjour.js]: done

**Review by alexander-akait — APPROVED** [review]

**Review by snitin315 — APPROVED** [review]

**Pull request merged**